### ISSUE TITLE
[unstable] rebuild fails on current unstable channel

### ISSUE THREAD

**Issue by matthiasbeyer** [NixOS/nixpkgs#8325]
I just saw that there was a unstable channel update, so I did:

```
nix-channel --update
nixos-rebuidl test
```

and it failed (again) with:

```
[...]
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/share/vim/vim74/tools/efm_perl.pl
not an ELF executable
gzipping man pages in /nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516
patching script interpreter paths in /nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/bin/gvimtutor: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/bin/vimtutor: interpreter directive changed from " /bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/share/vim/vim74/macros/less.sh: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/share/vim/vim74/tools/ref: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/share/vim/vim74/tools/vimm: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
/nix/store/z22kaw3hcvbjchdnxn9f5x1mh8nbfaj8-vim_configurable-7.4.516/share/vim/vim74/tools/vimspell.sh: interpreter directive changed from "/bin/sh" to "/nix/store/dbxpkswwc7rh6g1iy6dwqklzw39hihb1-bash-4.3-p33/bin/sh  "
basename: invalid option -- 'w'
Try 'basename --help' for more information.
...............................................................................................................
Symlinking /nix/store/5jnfr7xzg0prz618l3nayibi51jpj9ym-perl-5.20.2
find: `/nix/store/5jnfr7xzg0prz618l3nayibi51jpj9ym-perl-5.20.2/share/info': No such file or directory
find: `/nix/store/5jnfr7xzg0prz618l3nayibi51jpj9ym-perl-5.20.2/share/info': No such file or directory
.
cp: cannot stat '/nix/store/5jnfr7xzg0prz618l3nayibi51jpj9ym-perl-5.20.2/libexec': No such file or directory
updmap is using the following updmap.cfg files (in precedence order):
  /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg
  /nix/store/rm5fkmlf2y134sdzz0jxfjkbjzv6d2ra-texlive-core-2014/texmf-config/web2c/updmap.cfg
  /nix/store/rm5fkmlf2y134sdzz0jxfjkbjzv6d2ra-texlive-core-2014/texmf-dist/web2c/updmap.cfg
updmap is using the following updmap.cfg file for writing changes:
  /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg
Missing map files found, disabling
        bkaiu.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        bsmiu.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        dgj.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        dmj.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        gbsnu.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        gkaiu.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        lm.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        mc2j.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        mcj.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        mr2j.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        mrj.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        original-context-symbol.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qag.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qbk.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qcr.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qcs.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qhv.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qpl.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qtm.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        qzc.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
        tipa.map (in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg)
in /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg
Do you really want to continue (y/N)? answer =y=
/nix/store/rm5fkmlf2y134sdzz0jxfjkbjzv6d2ra-texlive-core-2014/bin/updmap --syncwithtrees finished.
Now you need to run updmap normally to recreate map files.
updmap is using the following updmap.cfg files (in precedence order):
  /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg
  /nix/store/rm5fkmlf2y134sdzz0jxfjkbjzv6d2ra-texlive-core-2014/texmf-config/web2c/updmap.cfg
  /nix/store/rm5fkmlf2y134sdzz0jxfjkbjzv6d2ra-texlive-core-2014/texmf-dist/web2c/updmap.cfg
updmap is using the following updmap.cfg file for writing changes:
  /nix/store/w7zsc9wf6qldy46h3h6sirblwvrwr4xp-TeXLive-linkdir/texmf-config/web2c/updmap.cfg
cp: cannot stat 'COPYING': No such file or directory
cp: cannot stat 'LICENSE': No such file or directory
cp: cannot stat 'DISTRIBUTION': No such file or directory
cp: cannot stat 'LEGAL': No such file or directory
cp: cannot stat 'README': No such file or directory
cp: cannot stat 'AUTHORS': No such file or directory
cp: cannot stat 'ChangeLog': No such file or directory
cp: cannot stat 'CHANGES': No such file or directory
cp: cannot stat 'LICENCE': No such file or directory
cp: cannot stat 'COPYRIGHT': No such file or directory
cp: cannot stat 'doc': No such file or directory
cannot build derivation ‘/nix/store/w6q8j77pwp9xi3g8pxv7429p9zqfmvlw-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/n9nr58y0ax9pahq9hzypr746iki5r6dw-dbus-conf.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/2b4nfqj5r1s15n7kpy95jqxw6y4c7yic-system-crontab.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/881lk17mrf163fm8wgil613d68jk9s67-unit-polkit.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/sza8saxdhsj4l5i66cyyps2blf7kg366-unit-dbus.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/xn5i207zr9pjim3kswqladv6sqq9p75l-unit-script.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ighw9igpaxw94i2z07pfnnalfhs0v53z-unit-cron.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/mpsvsm9n4j97ymcsd6lq6kw08mn6vi1n-system-units.drv’: 3 dependencies couldn't be built
cannot build derivation ‘/nix/store/cdqd7c80scf9mpl029gypx761qifvgqs-etc.drv’: 2 dependencies couldn't be built
cannot build derivation ‘/nix/store/bckakji15mcjix3abin80900zh6p62lk-nixos-15.06pre64030.e1af50c.drv’: 2 dependencies couldn't be built
error: build of ‘/nix/store/bckakji15mcjix3abin80900zh6p62lk-nixos-15.06pre64030.e1af50c.drv’ failed
```

It gets a bit frustrating ...

**Comment by vcunat** [comment]
Can you specify which derivation actually failed? I can't see that from your jumble of outputs, except that it's something that's in your system path. (I personally use ``-Q`` to get a better overview of what actually fails, and then I inspect logs of individual derivations separately.)

**Comment by matthiasbeyer** [comment]
That's my output with `-Q`:

```
$ sudo nixos-rebuild test -Q
[sudo] Passwort für m: 
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/bg4hndxhzlfma5iy2nndd3ah4x76s9mr-bmon-3.6.drv
  /nix/store/w6q8j77pwp9xi3g8pxv7429p9zqfmvlw-system-path.drv
  /nix/store/2b4nfqj5r1s15n7kpy95jqxw6y4c7yic-system-crontab.drv
  /nix/store/881lk17mrf163fm8wgil613d68jk9s67-unit-polkit.service.drv
  /nix/store/xn5i207zr9pjim3kswqladv6sqq9p75l-unit-script.drv
  /nix/store/ighw9igpaxw94i2z07pfnnalfhs0v53z-unit-cron.service.drv
  /nix/store/n9nr58y0ax9pahq9hzypr746iki5r6dw-dbus-conf.drv
  /nix/store/sza8saxdhsj4l5i66cyyps2blf7kg366-unit-dbus.service.drv
  /nix/store/mpsvsm9n4j97ymcsd6lq6kw08mn6vi1n-system-units.drv
  /nix/store/cdqd7c80scf9mpl029gypx761qifvgqs-etc.drv
  /nix/store/bckakji15mcjix3abin80900zh6p62lk-nixos-15.06pre64030.e1af50c.drv
building path(s) ‘/nix/store/dj5sx4hlrpfwq8wzdlq9r3a5bi6wxhpp-bmon-3.6’
builder for ‘/nix/store/bg4hndxhzlfma5iy2nndd3ah4x76s9mr-bmon-3.6.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/w6q8j77pwp9xi3g8pxv7429p9zqfmvlw-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/n9nr58y0ax9pahq9hzypr746iki5r6dw-dbus-conf.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/2b4nfqj5r1s15n7kpy95jqxw6y4c7yic-system-crontab.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/881lk17mrf163fm8wgil613d68jk9s67-unit-polkit.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/sza8saxdhsj4l5i66cyyps2blf7kg366-unit-dbus.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/xn5i207zr9pjim3kswqladv6sqq9p75l-unit-script.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/ighw9igpaxw94i2z07pfnnalfhs0v53z-unit-cron.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/mpsvsm9n4j97ymcsd6lq6kw08mn6vi1n-system-units.drv’: 3 dependencies couldn't be built
cannot build derivation ‘/nix/store/cdqd7c80scf9mpl029gypx761qifvgqs-etc.drv’: 2 dependencies couldn't be built
cannot build derivation ‘/nix/store/bckakji15mcjix3abin80900zh6p62lk-nixos-15.06pre64030.e1af50c.drv’: 2 dependencies couldn't be built
error: build of ‘/nix/store/bckakji15mcjix3abin80900zh6p62lk-nixos-15.06pre64030.e1af50c.drv’ failed
```

So it seems this is an issue of `bmon`. Thank you for providing me a way to see what fails more easily, though!

**Comment by matthiasbeyer** [comment]
Works when removing `bmon` from my configuration, thanks!

**Comment by vcunat** [comment]
I just fixed bmon build on master in a9c339283.

**Comment by matthiasbeyer** [comment]
:+1:
